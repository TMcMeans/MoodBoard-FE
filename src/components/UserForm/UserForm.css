/*COMPONENT STYLES*/
.user-form {
  color: #fff;
  background: linear-gradient(-45deg, #ee7752, #e73c7e, #23a6d5, #23d5ab);
  background-size: 400% 400%;
  height: 100vh;
  
  animation: Gradient 15s ease infinite;
  -webkit-animation: Gradient 15s ease infinite;
  -moz-animation: Gradient 15s ease infinite;
}

/*BACKGROUND ANIMATION*/
@-webkit-keyframes Gradient {
  0% {
    background-position: 0% 50%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0% 50%;
  }
}

@keyframes Gradient {
  0% {
    background-position: 0% 50%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0% 50%;
  }
}

@-moz-keyframes Gradient {
  0% {
    background-position: 0% 50%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0% 50%;
  }
}

.form-title {
  font-size: 2.5rem;
}

/*FORM STYLES*/
.user-info {
  align-items: center;
  background: #2A2A2A;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  height: 290px;
  margin: 0 auto;
  padding: 35px 0 0;
  width: 80%;
}

.user-input {
  font-size: 1.5em;
  height: 40px;
  margin: 0 auto 20px auto;
  padding-left: 10px;
  width: 75%;
}

.form-btn {
  font-size: 1.4rem;
  height: 3rem;
  line-height: 1rem;
  margin-top: 1rem;
  width: 6rem;
}

/*LINK*/
.form-msg {
  margin: auto;
  margin-bottom: 0;
  margin-top: 5px;
  width: 80%;
}

.login-link {
  color: #ffffff;
}

/*INFO BUTTON*/
.info-btn {
  background-color: #2A2A2A;
  border: none;
  border-radius: 50%;
  color: #ffffff;
  font-size: 2rem;
  font-weight: bold;
  height: 3rem;
  position: relative;
    left: 10rem;
  width: 3rem;
}
