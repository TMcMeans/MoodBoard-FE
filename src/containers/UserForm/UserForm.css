/*COMPONENT STYLES*/
.user-form {
  height: 100vh;
  color: #fff;
  background: linear-gradient(-45deg, #ee7752, #e73c7e, #23a6d5, #23d5ab);
  background-size: 400% 400%;
  -webkit-animation: Gradient 15s ease infinite;
  -moz-animation: Gradient 15s ease infinite;
  animation: Gradient 15s ease infinite;
}

@-webkit-keyframes Gradient {
  0% {
    background-position: 0% 50%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0% 50%;
  }
}

@keyframes Gradient {
  0% {
    background-position: 0% 50%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0% 50%;
  }
}

@-moz-keyframes Gradient {
  0% {
    background-position: 0% 50%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0% 50%;
  }
}

.form-title {
  font-size: 2.5rem;
}

/*FORM STYLES*/
.user-info {
  background: #2A2A2A;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  height: 290px;
  align-items: center;
  margin: 0 auto;
  padding: 35px 0 0;
  width: 80%;
}

.user-input {
  width: 75%;
  height: 40px;
  font-size: 1.5em;
  padding-left: 10px;
  margin: 0 auto 20px auto;
}

.form-btn {
  height: 3rem;
  line-height: 1rem;
  margin-top: 1rem;
}

/*LINK*/
.login-link {
  color: #ffffff;
}

/*INFO BUTTON*/
.info-btn {
  display: none;
  color: #ffffff;
  background-color: #2A2A2A;
  border: none;
  border-radius: 50%;
  font-size: 2rem;
  font-weight: bold;
  height: 3rem;
  position: relative;
    top: 3rem;
    left: 10rem;
  width: 3rem;
}

