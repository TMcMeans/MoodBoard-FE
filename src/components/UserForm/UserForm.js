import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import Logo from '../../components/Logo/Logo';

import './UserForm.css';

class UserForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      email: ''
    };
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value })
  }

  render() {
    const { name, email } = this.state;
    const { type } = this.props;
    let text;
    let switchText;
    let formBtn;

    if (type === 'login') {
      text = 'log in';
      switchText = 'First time at MoodBoard? Sign up ';
      formBtn = (
        <Link to="/signup" className="login-link">
          here
        </Link>
      )
    } else if (type === 'signup') {
      text = 'sign up';
      switchText = 'Already a user? Log in ';
      formBtn = (
        <Link to="/" className="login-link">
          here
        </Link>
      )
    }

    return (
      <div className="user-form">
        <Logo />
        <h1 className="form-title">{text}!</h1>
        <form action="" className="user-info">
          <input
            name="name"
            type="text"
            placeholder="name"
            className="user-input"
            value={name}
            onChange={this.handleChange}
          />
          <input
            name="email"
            type="text"
            placeholder="email"
            className="user-input"
            value={email}
            onChange={this.handleChange}
          />
          <input
            type="password"
            placeholder="password"
            className="user-input"
          />
          <Link to="/home">
            <button className="standard-btn light form-btn">
              {text}
            </button>
          </Link>
        </form>
        <h2 className="form-msg">
          {switchText}
          {formBtn}
        .</h2>
        <Link to='/info'><button className="info-btn">i</button></Link>
      </div>
    );
  }
}

export default UserForm;
