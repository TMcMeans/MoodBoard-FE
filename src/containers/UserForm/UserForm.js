import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import Logo from '../../components/Logo/Logo';

import './UserForm.css';

class UserForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      email: ''
    };
  }

  render() {
    const { name, email } = this.state;
    const { type } = this.props;
    let text;
    let switchText;
    let formBtn;

    if (type === 'login') {
      text = 'log in';
      switchText = 'First time at MoodBoard? Sign up ';
      formBtn = (
        <Link to="/signup" className="login-link">
          here
        </Link>
      )
    } else if (type === 'signup') {
      text = 'sign up';
      switchText = 'Already a user? Log in ';
      formBtn = (
        <Link to="/" className="login-link">
          here
        </Link>
      )
    }

    return (
      <div className="user-form">
        <Logo />
        <h1 className="form-title">{text}!</h1>
        <form action="" className="user-info">
          <input
            type="text"
            placeholder="name"
            className="user-input"
            value={this.state.name}
          />
          <input
            type="text"
            placeholder="email"
            className="user-input"
            value={this.state.email}
          />
          <input
            type="password"
            placeholder="password"
            className="user-input"
          />
          <Link to="/home">
            <button className="standard-btn light form-btn">
              {text}
            </button>
          </Link>
        </form>
        <h2>
          {switchText}
          {formBtn}
          .</h2>
        <button className="info-btn">i</button>
      </div>
    );
  }
}

export default UserForm;
