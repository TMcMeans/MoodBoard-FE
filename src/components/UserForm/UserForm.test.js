import React from 'react';
import { shallow } from 'enzyme';
import UserForm from './UserForm';

describe('UserForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<UserForm />);
  });

  it('should render a snapshot with correct data passed in', () => {
    wrapper = shallow(<UserForm type={'login'} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a snapshot with correct data passed in', () => {
    wrapper = shallow(<UserForm type={'signup'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
