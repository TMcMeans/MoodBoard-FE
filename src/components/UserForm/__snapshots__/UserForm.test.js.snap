// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm should render a snapshot with correct data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserForm
    type="login"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo />,
        <h1
          className="form-title"
        >
          log in
          !
        </h1>,
        <form
          action=""
          className="user-info"
        >
          <input
            className="user-input"
            placeholder="name"
            type="text"
            value=""
          />
          <input
            className="user-input"
            placeholder="email"
            type="text"
            value=""
          />
          <input
            className="user-input"
            placeholder="password"
            type="password"
          />
          <Link
            replace={false}
            to="/home"
          >
            <button
              className="standard-btn light form-btn"
            >
              log in
            </button>
          </Link>
        </form>,
        <h2>
          Already a user? Log in 
          <Link
            className="login-link"
            replace={false}
            to="/"
          >
            here
          </Link>
          .
        </h2>,
        <Link
          replace={false}
          to="/info"
        >
          <button
            className="info-btn"
          >
            i
          </button>
        </Link>,
      ],
      "className": "user-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "log in",
            "!",
          ],
          "className": "form-title",
        },
        "ref": null,
        "rendered": Array [
          "log in",
          "!",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "",
          "children": Array [
            <input
              className="user-input"
              placeholder="name"
              type="text"
              value=""
            />,
            <input
              className="user-input"
              placeholder="email"
              type="text"
              value=""
            />,
            <input
              className="user-input"
              placeholder="password"
              type="password"
            />,
            <Link
              replace={false}
              to="/home"
            >
              <button
                className="standard-btn light form-btn"
              >
                log in
              </button>
            </Link>,
          ],
          "className": "user-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "user-input",
              "placeholder": "name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "user-input",
              "placeholder": "email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "user-input",
              "placeholder": "password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="standard-btn light form-btn"
              >
                log in
              </button>,
              "replace": false,
              "to": "/home",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "log in",
                "className": "standard-btn light form-btn",
              },
              "ref": null,
              "rendered": "log in",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Already a user? Log in ",
            <Link
              className="login-link"
              replace={false}
              to="/"
            >
              here
            </Link>,
            ".",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Already a user? Log in ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "here",
              "className": "login-link",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "here",
            "type": [Function],
          },
          ".",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="info-btn"
          >
            i
          </button>,
          "replace": false,
          "to": "/info",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "i",
            "className": "info-btn",
          },
          "ref": null,
          "rendered": "i",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <h1
            className="form-title"
          >
            log in
            !
          </h1>,
          <form
            action=""
            className="user-info"
          >
            <input
              className="user-input"
              placeholder="name"
              type="text"
              value=""
            />
            <input
              className="user-input"
              placeholder="email"
              type="text"
              value=""
            />
            <input
              className="user-input"
              placeholder="password"
              type="password"
            />
            <Link
              replace={false}
              to="/home"
            >
              <button
                className="standard-btn light form-btn"
              >
                log in
              </button>
            </Link>
          </form>,
          <h2>
            Already a user? Log in 
            <Link
              className="login-link"
              replace={false}
              to="/"
            >
              here
            </Link>
            .
          </h2>,
          <Link
            replace={false}
            to="/info"
          >
            <button
              className="info-btn"
            >
              i
            </button>
          </Link>,
        ],
        "className": "user-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "log in",
              "!",
            ],
            "className": "form-title",
          },
          "ref": null,
          "rendered": Array [
            "log in",
            "!",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "action": "",
            "children": Array [
              <input
                className="user-input"
                placeholder="name"
                type="text"
                value=""
              />,
              <input
                className="user-input"
                placeholder="email"
                type="text"
                value=""
              />,
              <input
                className="user-input"
                placeholder="password"
                type="password"
              />,
              <Link
                replace={false}
                to="/home"
              >
                <button
                  className="standard-btn light form-btn"
                >
                  log in
                </button>
              </Link>,
            ],
            "className": "user-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "user-input",
                "placeholder": "name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "user-input",
                "placeholder": "email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "user-input",
                "placeholder": "password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="standard-btn light form-btn"
                >
                  log in
                </button>,
                "replace": false,
                "to": "/home",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "log in",
                  "className": "standard-btn light form-btn",
                },
                "ref": null,
                "rendered": "log in",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Already a user? Log in ",
              <Link
                className="login-link"
                replace={false}
                to="/"
              >
                here
              </Link>,
              ".",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Already a user? Log in ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "here",
                "className": "login-link",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "here",
              "type": [Function],
            },
            ".",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="info-btn"
            >
              i
            </button>,
            "replace": false,
            "to": "/info",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "i",
              "className": "info-btn",
            },
            "ref": null,
            "rendered": "i",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`UserForm should render a snapshot with correct data passed in 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserForm
    type="signup"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo />,
        <h1
          className="form-title"
        >
          sign up
          !
        </h1>,
        <form
          action=""
          className="user-info"
        >
          <input
            className="user-input"
            placeholder="name"
            type="text"
            value=""
          />
          <input
            className="user-input"
            placeholder="email"
            type="text"
            value=""
          />
          <input
            className="user-input"
            placeholder="password"
            type="password"
          />
          <Link
            replace={false}
            to="/home"
          >
            <button
              className="standard-btn light form-btn"
            >
              sign up
            </button>
          </Link>
        </form>,
        <h2>
          Already a user? Log in 
          <Link
            className="login-link"
            replace={false}
            to="/"
          >
            here
          </Link>
          .
        </h2>,
        <Link
          replace={false}
          to="/info"
        >
          <button
            className="info-btn"
          >
            i
          </button>
        </Link>,
      ],
      "className": "user-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "sign up",
            "!",
          ],
          "className": "form-title",
        },
        "ref": null,
        "rendered": Array [
          "sign up",
          "!",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "",
          "children": Array [
            <input
              className="user-input"
              placeholder="name"
              type="text"
              value=""
            />,
            <input
              className="user-input"
              placeholder="email"
              type="text"
              value=""
            />,
            <input
              className="user-input"
              placeholder="password"
              type="password"
            />,
            <Link
              replace={false}
              to="/home"
            >
              <button
                className="standard-btn light form-btn"
              >
                sign up
              </button>
            </Link>,
          ],
          "className": "user-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "user-input",
              "placeholder": "name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "user-input",
              "placeholder": "email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "user-input",
              "placeholder": "password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="standard-btn light form-btn"
              >
                sign up
              </button>,
              "replace": false,
              "to": "/home",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "sign up",
                "className": "standard-btn light form-btn",
              },
              "ref": null,
              "rendered": "sign up",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Already a user? Log in ",
            <Link
              className="login-link"
              replace={false}
              to="/"
            >
              here
            </Link>,
            ".",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Already a user? Log in ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "here",
              "className": "login-link",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "here",
            "type": [Function],
          },
          ".",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="info-btn"
          >
            i
          </button>,
          "replace": false,
          "to": "/info",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "i",
            "className": "info-btn",
          },
          "ref": null,
          "rendered": "i",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <h1
            className="form-title"
          >
            sign up
            !
          </h1>,
          <form
            action=""
            className="user-info"
          >
            <input
              className="user-input"
              placeholder="name"
              type="text"
              value=""
            />
            <input
              className="user-input"
              placeholder="email"
              type="text"
              value=""
            />
            <input
              className="user-input"
              placeholder="password"
              type="password"
            />
            <Link
              replace={false}
              to="/home"
            >
              <button
                className="standard-btn light form-btn"
              >
                sign up
              </button>
            </Link>
          </form>,
          <h2>
            Already a user? Log in 
            <Link
              className="login-link"
              replace={false}
              to="/"
            >
              here
            </Link>
            .
          </h2>,
          <Link
            replace={false}
            to="/info"
          >
            <button
              className="info-btn"
            >
              i
            </button>
          </Link>,
        ],
        "className": "user-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "sign up",
              "!",
            ],
            "className": "form-title",
          },
          "ref": null,
          "rendered": Array [
            "sign up",
            "!",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "action": "",
            "children": Array [
              <input
                className="user-input"
                placeholder="name"
                type="text"
                value=""
              />,
              <input
                className="user-input"
                placeholder="email"
                type="text"
                value=""
              />,
              <input
                className="user-input"
                placeholder="password"
                type="password"
              />,
              <Link
                replace={false}
                to="/home"
              >
                <button
                  className="standard-btn light form-btn"
                >
                  sign up
                </button>
              </Link>,
            ],
            "className": "user-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "user-input",
                "placeholder": "name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "user-input",
                "placeholder": "email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "user-input",
                "placeholder": "password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="standard-btn light form-btn"
                >
                  sign up
                </button>,
                "replace": false,
                "to": "/home",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "sign up",
                  "className": "standard-btn light form-btn",
                },
                "ref": null,
                "rendered": "sign up",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Already a user? Log in ",
              <Link
                className="login-link"
                replace={false}
                to="/"
              >
                here
              </Link>,
              ".",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Already a user? Log in ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "here",
                "className": "login-link",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "here",
              "type": [Function],
            },
            ".",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="info-btn"
            >
              i
            </button>,
            "replace": false,
            "to": "/info",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "i",
              "className": "info-btn",
            },
            "ref": null,
            "rendered": "i",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
